rackcity/models/asset.py

```diff
@@ -15,6 +16,23 @@ def get_next_available_asset_number():
         except ObjectDoesNotExist:
             return asset_number
 
+def get_assets_for_cp(change_plan=None):
+    """
+    If a change plan is specified, returns Asset query and AssetCP query,
+    where any assets modified in the change plan are in the AssetCP query but
+    removed from the Asset query. If no change plan is specified, returns
+    Asset query of all assets on master.
+    """
+    assets = Asset.objects.all()
+    if change_plan is None:
+        return (assets, None)
+    assetsCP = AssetCP.objects.filter(change_plan=change_plan)
+    for assetCP in assetsCP:
+        if (assetCP.related_asset) and (assetCP.related_asset) in assets:
+            assets = assets.filter(~Q(id=assetCP.related_asset.id))
+    # TODO remove any Asset or AssetCP that has been decommissioned in this CP
+    return (assets, assetsCP)
+
```

Comment:
i... will refactor so we dont have duplicated functions later. didnt want to mess up other stuff /deal w merge conflicts!

Reply:
refactoring this on a diff brnach!

Reply:
jk , fixed on this branch